src/security/acl_api.c

```diff
@@ -829,3 +873,170 @@ daos_ace_dump(struct daos_ace *ace, uint tabs)
 	print_all_flags(tabs + 1, ace);
 	print_all_perm_types(tabs + 1, ace);
 }
+
+static bool
+type_matches_principal_len(struct daos_ace *ace)
+{
+	bool len_exists;
+
+	len_exists = ace->dae_principal_len != 0;
+
+	return type_needs_name(ace->dae_principal_type) == len_exists;
+}
+
+static bool
+type_matches_flags(struct daos_ace *ace)
+{
+	bool flag_exists;
+
+	flag_exists = (ace->dae_access_flags & DAOS_ACL_FLAG_GROUP) != 0;
+
+	return type_is_group(ace->dae_principal_type) == flag_exists;
+}
+
+static bool
+principal_is_null_terminated(struct daos_ace *ace)
```

Comment:
i may not have understood the purpose of this function, but shouldn't the null termination be at dae_principal[dae_principal_len] considering dae_principal_len is the length of the string "strlen"?

Reply:
dae_principal_len may be longer than the actual length of the string, since we round the buffer size up for alignment purposes.

Reply:
I see. yea i didn't know that the len could be larger than actual size and there is an alignment required (probably due to cart transfer). thanks for explaining